pkg/gatewayserver/scheduling/sub_band.go

```diff
@@ -117,26 +135,74 @@ func (sb *SubBand) prioritizedDutyCycle(p ttnpb.TxSchedulePriority) float32 {
 	return sb.DutyCycle * ceiling
 }
 
-var errDutyCycle = errors.DefineResourceExhausted("duty_cycle", "utilization `{used}%` would be higher than the available `{usable}%` for priority `{priority}`")
+var (
+	errDutyCycle = errors.DefineResourceExhausted(
+		"duty_cycle",
+		"utilization `{used}%` would be higher than the available `{usable}%` for priority `{priority}`",
+	)
+	errBlocked = errors.DefineResourceExhausted(
+		"blocked",
+		"sub band is blocked for `{duration}`",
+	)
+)
+
+// checkDutyCycle verifies if the emission complies with the duty cycle limitations, based on the style.
+//
+// For the simple window style, it verifies if the emission will not exceed the usable amount of the duty
+// cycle in the [t + toa - window, t + toa] and [t, t + window] windows, where `t` is the start of the
+// emission and `toa` is the duration of the emission.
+//
+// For the blocking window style, in addition to the above mentioned check it verifies if the inverse of
+// the duty cycle has passed from the last emission.
+// Given a duty cycle of p%, and an emission of d time units at timestamp t, the duty cycle check will
+// allow the next transmission to occur only after the t + d * (100%/p%) timestamp.
```

Comment:
shouldn't this be ` t + d * ((100%/p%) -1)`?

Reply:
https://github.com/lorabasics/basicstation/blob/ba4f85d80a438a5c2b659e568cd2d0f0de08e5a7/src/s2e.c#L208-L212
https://github.com/lorabasics/basicstation/blob/ba4f85d80a438a5c2b659e568cd2d0f0de08e5a7/src/s2e.c#L375
https://github.com/lorabasics/basicstation/blob/ba4f85d80a438a5c2b659e568cd2d0f0de08e5a7/src/s2e.c#L-479-L480

The way I interpret it is that if a duration `d` can block the air only 1% of the time, the rest of 99% of the time should not be used either (in order to ensure that in this period we only used the air time only 1% of the time).

Reply:
If the transmission takes 3us at t=5us, then the waiting time should be 3 * 100 = 300us. No new transmissions until t=305us.In the [5us, 305us] we have used only 5us of airtime, which is 1%, and this is how BasicStation validates dutycycle - there are no sample windows.

Reply:
> t is the timestamp at the start of the transmission, not the end of it. So if we block [t, t + d * (100%/dc%)], we did take the duration into consideration.

Ok I see it now, this is indeed correct.